tests/context.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { cleanup, render } from '@testing-library/react'
+import { StateCreator } from '../src/index'
+import createContext from '../src/context'
+
+const consoleError = console.error
+afterEach(() => {
+  cleanup()
+  console.error = consoleError
+})
+
+type CounterState = {
+  count: number
+  inc: () => void
+}
+
+it('creates and uses context store', async () => {
+  const { Provider, useStore } = createContext<CounterState>()
+
+  const createParam: StateCreator<CounterState> = (set) => ({
+    count: 0,
+    inc: () => set((state) => ({ count: state.count + 1 })),
+  })
+
+  function Counter() {
+    const { count, inc } = useStore()
+    React.useEffect(inc, [inc])
+    return <div>count: {count}</div>
```

Comment:
but I wonder if this is properly typed.
```suggestion
    return <div>count: {count * 1}</div>
```
This can detect if `count` is inferred as number.